cognite_toolkit/_cdf_tk/commands/pull.py

```diff
@@ -799,47 +847,113 @@ def _update(
                         if placeholder in extra_content:
                             new_extra = new_extra.replace(variable.value, f"{{{{ {variable.key} }}}}")
                     extra_files[extra.path] = new_extra
-        return cls._replace(loaded, loaded_with_placeholder, item_write, value_by_placeholder)
+        return replacer.replace(loaded, loaded_with_placeholder, item_write)
 
-    @classmethod
-    def _replace(
-        cls,
-        loaded: dict[str, Any],
-        loaded_with_placeholder: dict[str, Any],
+
+class ResourceReplacer:
```

Comment:
Turned out to be more complex that the initial implementation. In particular how to deal with lists.